src/xpk/core/blueprint/blueprint_generator.py

```diff
@@ -354,6 +359,266 @@ def _get_a3_mega_blueprint_dependencies(
     )
     return deployment_files_path
 
+  def _get_a3_ultra_blueprint_dependencies(
+      self, blueprint_name: str, prefix: str = ""
+  ) -> str:
+    deployment_files_path = os.path.join(
+        self._get_storage_path(prefix), blueprint_name
+    )
+    shutil.copytree(
+        blueprint_dependencies_dir[a3ultra_device_type],
+        deployment_files_path,
+        dirs_exist_ok=True,
+    )
+    return deployment_files_path
+
+  def generate_a3_ultra_blueprint(
+      self,
+      project_id: str,
+      cluster_name: str,
+      blueprint_name: str,
+      region: str,
+      zone: str,
+      auth_cidr: str,
+      system_node_pool_machine_type: str,
+      reservation: Optional[str | None] = None,
+      static_node_count: int = 4,
+      prefix: str = "",
+      mtu_size: int = 8896,
+      system_node_pool_min_node_count: int = 2,
+      spot: bool = False,
+  ) -> BlueprintGeneratorOutput:
+    """Create A3 ultra blueprint.
+
+    Args:
+    Returns:
+      - Blueprint representing cluster toolkit blueprint
+    """
+
+    nccl_installer_path = (
+        f'$(ghpc_stage("{blueprint_name}"))/nccl-installer.yaml'
+    )
+    mglru_disable_path = f'$(ghpc_stage("{blueprint_name}"))/mlgru-disable.yaml'
+    net_0_id = f"{cluster_name}-a3u-net-0"
```

Comment:
nit: do we use these variable(s) for the names?

Reply:
yes, we use it in other place in use lsit